### ISSUE TITLE
no-undefined-types gives incorrect warnings since 51.0.5

### ISSUE THREAD

**Issue by simonnilsson** [gajus/eslint-plugin-jsdoc#1425]
## Expected behavior
I expect no warnings when using a defined type. The same code worked fine in v51.0.4 and earlier.

## Actual behavior
I get the following warnings:
```
test.js
  1:7  warning  'otherFile' is assigned a value but never used  no-unused-vars
  5:1  warning  The type 'otherFile.MyType' is undefined        jsdoc/no-undefined-types
```

## ESLint Config
```js
const globals = require('globals');
const js = require('@eslint/js');
const jsdoc = require('eslint-plugin-jsdoc');

module.exports = [
  js.configs.recommended,
  {
    plugins: {
      jsdoc,
    },
    languageOptions: {
      globals: {
        ...globals.node,
      },
    },
    rules: {
      'jsdoc/no-undefined-types': 1,
      'no-unused-vars': ['warn', { args: 'none' }],
    },
  },
];
```

## Example
other-file.js
```js
/**
 * @typedef {Object} MyType
 */
```

test.js
```js
const otherFile = require('./other-file');

/**
 * A function
 * @param {otherFile.MyType} a
 */
function f(a) {}

f();
```

## Environment

- Node version: 22
- ESLint version 9.30.0
- `eslint-plugin-jsdoc` version: 51.0.5 - 51.3.3

**Comment by simonnilsson** [comment]
@brettz9 Great! Thanks for the quick fix.
